### PR TITLE
feat: [CDS-38820]: provide props in PillToggle component so that consumers show disable warning icon and tooltip

### PR DESCRIPTION
**Pull request by chetanvekariya-harness**
JIRA - https://harness.atlassian.net/browse/CDS-38820

Design Link: https://www.figma.com/file/4cqisYyQGDjzz51LyDp0Wj/Config-as-Code?node-id=3568%3A27734

Design image: 
Screenshot 2022-06-13 at 5 39 10 PM

**After Dev**
Storybook screenshot:
Screenshot 2022-06-13 at 5 27 20 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- StyleLint: `retrigger stylelint`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- ImageSnapshot `retrigger ImageSnapshot`

### CODE REVIEW

**Review by vkbansal — DISMISSED** [review]

File: packages/uicore/src/components/PillToggle/PillToggle.tsx [hunk 1]
```diff
@@ -20,10 +27,56 @@ export interface PillToggleProps<T> {
...
+                <Text
+                  width={284}
+                  color={Color.GREY_0}
+                  margin={{ bottom: 'small' }}
+                  font={{ size: 'normal', weight: 'light' }}>
+                  The Visual Editor is disabled because of the errors in the YAML file.
+                </Text>
+                <Text width={284} color={Color.GREY_0} font={{ size: 'normal', weight: 'light' }}>
+                  Fix all the errors indicated in the YAML Editor to enable the Visual mode.
+                </Text>
```
Comment (upadhyay-an) [inline packages/uicore/src/components/PillToggle/PillToggle.tsx:70]: To support internationalisation, these strings should be passed down as prop.

File: packages/uicore/src/components/PillToggle/PillToggle.css [hunk 2]
```diff
@@ -23,6 +23,11 @@
     padding: 2px;
     border-radius: 100px;
     cursor: pointer;
+
+    .disableToggleReasonIcon {
+      margin-left: 2px;
+      margin-right: 2px;
```
Comment (upadhyay-an) [inline packages/uicore/src/components/PillToggle/PillToggle.css:29]: ```margin: 0 var(--spacing-tiny)```

File: packages/uicore/src/components/PillToggle/PillToggle.tsx (hunk 1 above)
Comment (chetanvekariya-harness) [reply packages/uicore/src/components/PillToggle/PillToggle.tsx:70]: The idea is to give some default UI here. Consumers should pass their own UI for tooltip ideally and that will handle our concern for strings.

File: packages/uicore/src/components/PillToggle/PillToggle.css (hunk 2 above)
Comment (chetanvekariya-harness) [reply packages/uicore/src/components/PillToggle/PillToggle.css]: Fixed.

**Review by upadhyay-an — APPROVED** [review]

**Pull request merged**